src/context/WalletProvider/NativeWallet/components/NativeCreate.tsx

```diff
@@ -83,36 +84,40 @@ export const NativeCreate = () => {
   }, [history, isLegacyWallet, mixpanel, vault])
 
   useEffect(() => {
+    if (vault) return
     ;(async () => {
       try {
-        // If the vault is already passed from the legacy wallet flow, use it.
-        const vault = isLegacyWallet ? location.state.vault : await getVault()
+        const vault = location.state?.vault ?? (await getVault())
```

Comment:
q: at this point, any reason to keep this an IIAFE vs. a react-query?

Reply:
Converted to useQuery in [d8091b3](https://github.com/shapeshift/web/pull/8885/commits/d8091b329778cd99a575de52a02ddf6598afd3ed)

Also had to wait for word init using a ref before revoking as there was now a race condition that appeared